### ISSUE TITLE
Event for when a tile has loaded

### ISSUE THREAD

**Issue by burn123** [AnalyticalGraphicsInc/cesium#5572]
Would like an event listener for when a slippy tile has loaded. I'm trying to keep track of our calls to a map tile server, since that API doesn't provide a counter for how many requests we've done. I've almost gotten something hacked together using the `_requestImagery` function of `ImageryLayer`, but it is hard to do without modifying the source code, especially since `_requestImagery` doesn't return a promise or take a callback. If it did, I could at least check for `ImageryState.Received`.

**Comment by icemagno** [comment]
What about this? I need to show a "Loaging..." every time I PAN / ZOOM the map and the tiles are changing.

**Comment by OmarShehata** [comment]
You can do this with the [Globe.tileLoadProgressEvent](https://cesiumjs.org/Cesium/Build/Documentation/Globe.html?classFilter=GLobe#tileLoadProgressEvent) which tells you how many tiles are remaining. I used that in this Cesium Viewer app on Glitch to show a loading indicator on the bottom left:

https://cesium-viewer.glitch.me/

Source here: https://glitch.com/edit/#!/cesium-viewer?path=README.md:1:0

**Comment by icemagno** [comment]
Yeah I'm using this already but only works when I add a new layer.  I need this  in every layer updates.  After the first load this event is not fired anymore.

**Comment by OmarShehata** [comment]
This event fires every time any new tile is loaded as shown [in this Sandcastle example](https://cesiumjs.org/Cesium/Apps/Sandcastle/index.html#c=ZY/NasNADIRfZfElNhSZnOOYQujN0EKgJ1/Wa8UVlVdhtXZIS9+9G7uF/uikkb6RmNkGMxNeMJi98XgxB1SaRnheZvnGLfIgPlryGDbFrvUrD+rQIwwsHUIkxkZs/xRkCKj6MKOPYPt+aRrSmNiQnybvIonPb7wW7603qZx4FUZgGb4W6clHscvuskrjlbFeuXsazxKimQLnAGXE8cw2opbd5F4xgtPFWZXfpqqn2VC/b7M/MdrMOLaqaXOamI/0hm1WV2Xif9k4RSI/PM4Y2F5vyMu2btYhAFRlkv9dUYQ7G35c/AQ). If this is not what you're seeing please post a code example we can run on Sandcastle to reproduce the issue.

**Comment by icemagno** [comment]
You're right. My mistake. All working as you said.

**Comment by icemagno** [comment]
Ok. It will be  nice to have a "per layer" tiles loading.  Just like this global event.
